vumi_http_proxy/http_proxy.py

```diff
@@ -23,7 +25,21 @@ def buildProtocol(self, addr):
 class CheckProxyRequest(proxy.ProxyRequest):
     def process(self):
         host, _, port = self.getAllHeaders()['host'].partition(':')
-        if host in self.channel.blacklist:
+        self.d = client.getHostByName(host)
+        self.d.addCallback(self.setIP, host)
+        self.d.addErrback(printError)
```

Comment:
If an error occurs, you probably still want to send an HTTP response back, so you might want write your own error handling method that logs the error and calls `self.write(...)` with some HTML to return explaining what happened and `self.finish()` to close the request that is being handled.